src/test/java/com/linecorp/armeria/server/http/healthcheck/HttpHealthCheckServiceTest.java

```diff
@@ -97,12 +104,37 @@ public void notHealthyWhenServerIsStopping() throws Exception {
         assertNotOk();
     }
 
-    private void assertNotOk() throws Exception {Promise<Object> promise = executor.newPromise();
+    private void assertNotOk() throws Exception {
         final DefaultHttpRequest req = new DefaultHttpRequest(HttpHeaders.of(HttpMethod.GET, "/"));
         req.close();
         final AggregatedHttpMessage res = service.serve(context, req).aggregate().get();
 
         assertEquals(HttpStatus.SERVICE_UNAVAILABLE, res.status());
         assertEquals("not ok", res.content().toStringAscii());
     }
+
+    @Test
+    public void defaultResponseIsNotChunked() throws Exception {
+        final ServerBuilder builder = new ServerBuilder();
+        builder.port(0, SessionProtocol.HTTP);
+        builder.serviceAt("/l7check", new HttpHealthCheckService());
+        final Server server = builder.build();
+        try {
+            server.start().join();
+
+            final int port = server.activePort().get().localAddress().getPort();
+            try (Socket s = new Socket(NetUtil.LOCALHOST, port)) {
+                s.setSoTimeout(10000);
+                final InputStream in = s.getInputStream();
+                final OutputStream out = s.getOutputStream();
+                out.write("GET /l7check HTTP/1.0\r\n\r\n".getBytes(StandardCharsets.US_ASCII));
+
+                // Should not be chunked.
+                assertThat(new String(ByteStreams.toByteArray(in))).isEqualTo(
+                        "HTTP/1.1 200 OK\r\ncontent-length: 2\r\n\r\nok");
```

Comment:
I think it's fine:

http://serverfault.com/questions/442960/nginx-ignoring-clients-http-1-0-request-and-respond-by-http-1-1